android/capacitor/src/main/java/com/getcapacitor/plugin/CapacitorCookies.java

```diff
@@ -31,26 +32,41 @@ public boolean isEnabled() {
         return pluginConfig.getBoolean("enabled", false);
     }
 
-    @JavascriptInterface
-    public String getCookies() {
-        String cookieString = cookieManager.getCookieString(null);
-        return (null == cookieString) ? "" : cookieString;
-    }
-
     @JavascriptInterface
     public void setCookie(String domain, String action) {
         cookieManager.setCookie(domain, action);
     }
 
     @PluginMethod
     public void getCookies(PluginCall call) {
-        String url = call.getString("url");
-        JSObject cookiesMap = new JSObject();
-        HttpCookie[] cookies = cookieManager.getCookies(url);
-        for (HttpCookie cookie : cookies) {
```

Comment:
This won't work since the HttpCookie[] is being parsed from `webkitCookieManager.getCookie(url);` which returns a string of simply `key/value` pairs... In other words, the HttpCookies don't return any of their attributes that are set.

Reply:
I know it does that, but man, does that ever feel like a bug.

Reply:
I'm not exactly sure why Android intentionally wants to hide the cookie metadata from being exposed, especially if they are returning all the cookies regardless if `httpOnly` is set or not... But that is the way they have had it set up since API 1 (https://developer.android.com/reference/android/webkit/CookieManager#getCookie(java.lang.String)) and there doesn't seem to be any other way to access the actual properties of the cookies

Reply:
Yeah, it's clearly an intentional choice. But still, it seems feels a "bug" they don't care about.